queue/queue.go

```diff
@@ -7,14 +7,29 @@ import (
 
 	"github.com/frain-dev/convoy"
 	"github.com/frain-dev/convoy/datastore"
+	"github.com/go-redis/redis/v8"
+	"github.com/vmihailenco/taskq/v3"
 )
 
 type Queuer interface {
 	io.Closer
 	Write(context.Context, convoy.TaskName, *datastore.EventDelivery, time.Duration) error
+	Consumer() taskq.QueueConsumer
```

Comment:
@Subomi @ogbanugot should we be returning external objects in our own interfaces?